plugin/src/main/java/com/playmonumenta/scriptedquests/quests/components/actions/dialog/DialogAllInOneEntry.java

```diff
@@ -0,0 +1,89 @@
+package com.playmonumenta.scriptedquests.quests.components.actions.dialog;
+
+import java.util.Map.Entry;
+import java.util.Set;
+
+import org.bukkit.entity.Entity;
+import org.bukkit.entity.Player;
+
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.playmonumenta.scriptedquests.Plugin;
+import com.playmonumenta.scriptedquests.quests.components.QuestPrerequisites;
+import com.playmonumenta.scriptedquests.utils.MessagingUtils;
+
+import net.kyori.adventure.text.Component;
+import net.kyori.adventure.text.TextReplacementConfig;
+import net.kyori.adventure.text.event.ClickEvent;
+import net.kyori.adventure.text.event.HoverEvent;
+
+public class DialogAllInOneEntry implements DialogBase {
+
+	// It needs an initialized value
+	private Component mComponent;
+	private String mNPCName;
+
+	public DialogAllInOneEntry(String npcName, JsonElement element) throws Exception {
+
+		JsonObject object = element.getAsJsonObject();
+		if (object == null) {
+			throw new Exception("dialog value is not an object!");
+		}
+
+		mNPCName = npcName;
+
+		Set<Entry<String, JsonElement>> entries = object.entrySet();
+		for (Entry<String, JsonElement> ent : entries) {
+			String key = ent.getKey();
+
+			JsonElement value = object.get(key);
+			if (value == null) {
+				throw new Exception("clickable_text value for key '" + key + "' is not parseable!");
+			}
+
+			if (!key.equals("hover_text") && !key.equals("click_action") && !key.equals("player_text") && !key.equals("actual_text")) {
+				throw new Exception("Unknown clickable_text key: " + key);
+			}
+
+
+			if (key.equals("click_action")) {
+				if (mComponent.clickEvent() != null) {
+					throw new Exception("There can only be one on click event!");
+				}
```

Comment:
JSON won't let you use the same key twice in the same object, so this check isn't actually needed. It's needed below though, since those two things have different keys (click_command and click_url)

Reply:
to clarify, you can just remove this entire block 
```java
				if (mComponent.clickEvent() != null) {
					throw new Exception("There can only be one on click event!");
				}```